Source/track.cpp

```diff
@@ -21,8 +21,53 @@ bool sgbIsWalking;
 
 } // namespace
 
+static bool RepeatLeftMouseAttackAction() //Fluffy
+{
+	if (!(lastLeftMouseButtonAction == MOUSEACTION_ATTACK || lastLeftMouseButtonAction == MOUSEACTION_ATTACK_MONSTERTARGET || lastLeftMouseButtonAction == MOUSEACTION_ATTACK_PLAYERTARGET) || pcurs != CURSOR_HAND || sgbMouseDown != CLICK_LEFT)
+		return false;
+
+	//Repeat action if it's been X duration since the attack or spell cast
+	Uint32 currentTime = SDL_GetTicks();
+	if (currentTime - lastLeftMouseButtonTime > 200) { //Check if it's been at least 200ms
+		if (lastLeftMouseButtonAction == MOUSEACTION_ATTACK) {
+			if (cursmx >= 0 && cursmx < MAXDUNX && cursmy >= 0 && cursmy < MAXDUNY) { 
+				if (Players[MyPlayerId]._pwtype == WT_RANGED)
+					NetSendCmdLoc(MyPlayerId, true, CMD_RATTACKXY, {cursmx, cursmy});
+				else
+					NetSendCmdLoc(MyPlayerId, true, CMD_SATTACKXY, {cursmx, cursmy});
+			}
+		} else if (lastLeftMouseButtonAction == MOUSEACTION_ATTACK_MONSTERTARGET && pcursmonst != -1) {
+			if (Players[MyPlayerId]._pwtype == WT_RANGED)
+				NetSendCmdParam1(true, CMD_RATTACKID, pcursmonst);
+			else
+				NetSendCmdParam1(true, CMD_ATTACKID, pcursmonst);
+		} else if (lastLeftMouseButtonAction == MOUSEACTION_ATTACK_PLAYERTARGET && pcursplr != -1 && !gbFriendlyMode) {
+			if (Players[MyPlayerId]._pwtype == WT_RANGED)
+				NetSendCmdParam1(true, CMD_RATTACKPID, pcursplr);
+			else
+				NetSendCmdParam1(true, CMD_ATTACKPID, pcursplr);
+		}
+	}
+	return true;
+}
+
+static bool RepeatRightMouseAction() //Fluffy
+{
+	if (!(lastRightMouseButtonAction == MOUSEACTION_SPELL || lastRightMouseButtonAction == MOUSEACTION_ATTACK) || pcurs != CURSOR_HAND || sgbMouseDown != CLICK_RIGHT)
+		return false;
+
+	//Repeat action if it's been X duration since the attack or spell cast
+	Uint32 currentTime = SDL_GetTicks();
+	if (currentTime - lastRightMouseButtonTime > 200) //Check if it's been at least 200ms

```

Comment:
Rather then 200ms you should probably use a multiple of `gnTickDelay` so that it scales correctly with game speed. `gnTickDelay * 4` would be the same as 200ms at normal game speed.

Reply:
Done